check-disk/lib/check-disk.go

```diff
@@ -0,0 +1,212 @@
+package checkdisk
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/jessevdk/go-flags"
+	"github.com/mackerelio/checkers"
+	gpud "github.com/shirou/gopsutil/disk"
+)
+
+var opts struct {
+	Warning  *string `short:"w" long:"warning" value-name:"N, N%" description:"Exit with WARNING status if less than N units or N% of disk are free"`
+	Critical *string `short:"c" long:"critical" value-name:"N, N%" description:"Exit with CRITICAL status if less than N units or N% of disk are free"`
+	Path     *string `short:"p" long:"path" value-name:"PATH" description:"Mount point or block device as emitted by the mount(8) command"`
```

Comment:
check_disk has `-R`/`-r` options to select paths by regexp pattern.  Do you have any idea how to handle regexp pattern? (Simply implement those options, accept patterns in `-p`, etc...)